packages/quantum-metric/src/data-layer-adapter.ts

```diff
@@ -0,0 +1,67 @@
+import {DataLayerAdapter} from "@evolv-integrations/analytics-adapter";
+
+const AVAILABLE_PARAMETERS_TO_READ = {
+    GET_CART_VALUE: 'cartValue'
+};
+
+export class QuantumDataLayerAdapter extends DataLayerAdapter {
+
+
+    constructor(
+        public listenersParameters: Record<string, string> = {},
+        public parametersToReadFromQuantum: Record<string, boolean> = {},
+        public readonly maxWaitTime = 5000
+    ) {
+        super(maxWaitTime);
+
+        const defaultParametersToReadFromQuantum: Record<string, boolean> = {};
+        defaultParametersToReadFromQuantum[AVAILABLE_PARAMETERS_TO_READ.GET_CART_VALUE] = true;
+
+        const defaultListenerParams = {
+            '-2': 'rageClick',
+            '-5': 'possibleFrustration',
+            '-7': 'slowApiCall',
+            '-9': 'frustratedSlowNavigation',
+            '-19': 'frozenUI',
+            '-22': 'longRunningSpinner'
+        };
+
+        this.listenersParameters = Object.assign(defaultListenerParams, listenersParameters);
+        this.parametersToReadFromQuantum = Object.assign(defaultParametersToReadFromQuantum, parametersToReadFromQuantum)
```

Comment:
For safety, I might use the ES6 spread syntax here. When targeting ES5, TypeScript will use its `__assign()` helper function rather than `Object.assign()` (which is not IE11 compatible).

Generally, we assume integrations will be processed downstream by the Bundler, which should transpile `Object.assign()` using Babel. But the browser script will not be processed by the Bundler.
```suggestion
        this.listenersParameters = { ...defaultListenerParams, ...listenersParameters };
        this.parametersToReadFromQuantum = { ...defaultParametersToReadFromQuantum, ...parametersToReadFromQuantum };
```